docs/aws/audit/codebuildmonitoring/rules/codebuild_project_restrict_cred_in_environment_remediation.mdx

```diff
@@ -1,7 +1,233 @@
+To prevent plaintext AWS credentials in environment variables for an AWS CodeBuild project using the AWS Management Console, follow these steps:
+
+1. **Navigate to CodeBuild:**
+   - Open the AWS Management Console.
+   - In the search bar, type "CodeBuild" and select it from the dropdown.
+
+2. **Select the Project:**
+   - In the CodeBuild dashboard, locate and select the CodeBuild project you want to configure.
+
+3. **Edit Environment Variables:**
+   - In the project details page, click on the "Edit" button.
+   - Scroll down to the "Environment" section.
+   - Under "Environment variables," ensure that no sensitive information such as AWS credentials (AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY) is set in plaintext.
+
+4. **Use AWS Secrets Manager or Parameter Store:**
+   - Instead of setting sensitive information directly in environment variables, use AWS Secrets Manager or AWS Systems Manager Parameter Store.
+   - Store your credentials in Secrets Manager or Parameter Store and reference them in your buildspec file or through the CodeBuild project configuration.
+
+By following these steps, you can ensure that plaintext AWS credentials are not set in environment variables for your CodeBuild project, enhancing the security of your build environment.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent plaintext AWS credentials in environment variables for an AWS CodeBuild project using the AWS CLI, follow these steps:
+
+1. **Create an IAM Role for CodeBuild:**
+   Ensure you have an IAM role with the necessary permissions for your CodeBuild project. This role will be used instead of embedding credentials in environment variables.
+
+   ```sh
+   aws iam create-role --role-name CodeBuildServiceRole --assume-role-policy-document file://trust-policy.json
+   ```
+
+   The `trust-policy.json` should contain the trust relationship policy allowing CodeBuild to assume the role.
+
+2. **Attach Policies to the IAM Role:**
+   Attach the necessary policies to the IAM role to grant it the required permissions.
+
+   ```sh
+   aws iam attach-role-policy --role-name CodeBuildServiceRole --policy-arn arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess
+   ```
+
+3. **Create or Update the CodeBuild Project Without Plaintext Credentials:**
+   When creating or updating the CodeBuild project, specify the IAM role and avoid setting plaintext credentials in environment variables.
+
+   ```sh
+   aws codebuild create-project --name MyProject \
+   --source type=GITHUB,location=https://github.com/my-repo/my-project \
+   --artifacts type=NO_ARTIFACTS \
+   --environment type=LINUX_CONTAINER,image=aws/codebuild/standard:4.0,computeType=BUILD_GENERAL1_SMALL \
+   --service-role arn:aws:iam::123456789012:role/CodeBuildServiceRole
+   ```
+
+4. **Use AWS Secrets Manager or Parameter Store for Sensitive Data:**
+   If you need to use sensitive data, store it in AWS Secrets Manager or AWS Systems Manager Parameter Store and reference it in your buildspec file.
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>